packages/contracts/scripts/runner/runner.ts

```diff
@@ -1,35 +1,96 @@
 import axios from "axios";
 import { spawn } from "child_process";
-import { RPC_LIST, RPC_DELAY, REQ_BODY, RESP_STATUS } from "./conf";
+import { performance } from "node:perf_hooks";
+import { RETRY_COUNT, RETRY_DELAY, RPC_BODY, RPC_HEADER, RPC_LIST } from "./conf";
 
-const getRandom = async (array: string[]) => {
-  const length = array == null ? 0 : array.length;
-  return length ? array[Math.floor(Math.random() * length)] : undefined;
+type LatencyObjType = {
+  latency: number;
+  baseURL: string;
 };
 
-const getUrl = async () => {
-  let isRPC = false;
+const RPCLatency_Array: LatencyObjType[] = [];
 
-  while (!isRPC) {
+const sortByLatency = () => {
```

Comment:
This is dope, but couldn't we just do something like a `Promise.race` and take the first successful call so that we don't have to wait for everybody's response?

Reply:
> This is dope, but couldn't we just do something like a `Promise.race` and take the first successful call so that we don't have to wait for everybody's response?

`Promise.race` exits on the first settled promise, it can be either `resolved` or `rejected`.
![image](https://user-images.githubusercontent.com/109317559/210354770-d934d263-31fb-43b5-a17b-d9098f5c8bc4.png)

@pavlovcik

Reply:
> Promise.race exits on the first settled promise, it can be either resolved or rejected.

I understand that. That's why i said "like a `Promise.race` but I think you understand the general idea I'm going for!

Reply:
> > Promise.race exits on the first settled promise, it can be either resolved or rejected.
> 
> I understand that. That's why i said "like a `Promise.race`" but I think you understand the general idea I'm going for! If its too complicated then don't worry about it, but I presume that there is some type of npm package for this.

I wanted to go with the `Promise.any()` which exits on first resolved promise. 
But it requires the compilation version set to `es2021`, I couldn't find any `tsconfig.json`,
Should I commit one along with the changes?
@pavlovcik

Reply:
I found one in `packages/dapp/tsconfig.json`

I'm not sure how yarn workspaces handles this but please feel free to consolidate tsconfig.json configs correctly in a single location if that is how yarn workspaces handles this, or feel free to make a new tsconfig if thats the correct way to handle it.

Reply:
pushed the changes @pavlovcik